pkg/model/model.go

```diff
@@ -49,6 +49,11 @@ type Model struct {
 	docCfg *ackgenconfig.DocumentationConfig
 }
 
+type Nested struct {
+	arr    []string
+	Config *ackgenconfig.FieldConfig
+}
```

Comment:
i know this is working. Pretty sure we can find a better way to represent this data.. (i think we are copying a lot `arr`)

Reply:
I didn't actually get what you mean by copying `arr` data a lot. It is just defined once to store the names of nested fields and is then iterated over to check the nested fields type and whether they are a part of their parent field. But yeah I'm open to looking for a better way to represent the data.